src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +46,143 @@ public void editPerson() {
         editPersonDialog.enterNewValues(newAlice);
         editPersonDialog.pressEnter();
 
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
-        //TODO: write a more elegant function that uses assertEquals but compare two different types of objects
+        mainGui.sleep(1, TimeUnit.SECONDS);
+
+        //Confirm pending state correctness
+        assertTrue(alicePersonCard.isPendingStateRootVisible());
+        assertTrue(alicePersonCard.isPendingStateLabelVisible());
+        assertFalse(alicePersonCard.isPendingStateProgressIndicatorVisible());
+        assertTrue(alicePersonCard.getPendingStateLabel().equals("Editing"));
```

Comment:
Lump these into a method in the GuiHandle. e.g. 
`assertTrue(alicePersonCard.isShowingGracePeriod("Editing")`

It's best when we can write test at higher level abstractions.